api/dps/server_test.go

```diff
@@ -38,6 +38,69 @@ func TestNewServer(t *testing.T) {
 	assert.Equal(t, index, s.index)
 }
 
+func TestServerGetFirst(t *testing.T) {
+
+	var (
+		testHeight = uint64(128)
+	)
+
+	vectors := []struct {
+		description string
+
+		mockHeight uint64
+		mockErr    error
+
+		wantRes *GetFirstResponse
+
+		checkErr assert.ErrorAssertionFunc
+	}{
+		{
+			description: "happy case",
+
+			mockHeight: testHeight,
+			mockErr:    nil,
+
+			wantRes: &GetFirstResponse{
+				Height: testHeight,
+			},
+
+			checkErr: assert.NoError,
+		},
+		{
+			description: "error case",
+
+			mockHeight: testHeight,
+			mockErr:    errors.New("dummy error"),
+
+			wantRes: nil,
+
+			checkErr: assert.Error,
+		},
+	}
+
+	for _, vector := range vectors {
+		vector := vector
+		t.Run(vector.description, func(t *testing.T) {
+			t.Parallel()
+
+			index := &mocks.Reader{}
+			s := Server{index: index}
+
+			index.FirstFunc = func() (uint64, error) {
+				return vector.mockHeight, vector.mockErr
+			}
+
+			req := &GetFirstRequest{}
+
+			gotRes, gotErr := s.GetFirst(context.Background(), req)
+			vector.checkErr(t, gotErr)
+			if gotErr == nil {
+				assert.Equal(t, vector.wantRes, gotRes)
+			}
```

Comment:
This `if` statement should not be necessary. If there's an error we would expect `gotRes` to be `nil`, no?

Reply:
According to Go best practices, we shouldn't make any assumptions about values that are returned when `err != nil`. It could be `nil`, it could be a partial result, etc. So we shouldn't test it.